unstructured/ingest/connector/astra.py

```diff
@@ -0,0 +1,114 @@
+
+
+@dataclass
+class SimpleAstraDBConfig(BaseConnectorConfig):
+    access_config: AstraDBAccessConfig
+    collection_name: str
+    embedding_dimension: int
+
+
+@dataclass
+class AstraDBWriteConfig(WriteConfig):
+    batch_size: int = 20
+
+
+@dataclass
+class AstraDBDestinationConnector(BaseDestinationConnector):
+    write_config: AstraDBWriteConfig
+    connector_config: SimpleAstraDBConfig
+    _astra_db: t.Optional["AstraDB"] = field(init=False, default=None)
+    _astra_db_collection: t.Optional["AstraDBCollection"] = field(init=False, default=None)
+
+    def to_dict(self, **kwargs):
+        """
+        The _astra_db_collection variable in this dataclass breaks deepcopy due to:
+        TypeError: cannot pickle '_thread.lock' object
+        When serializing, remove it, meaning client data will need to be reinitialized
+        when deserialized
+        """
+        self_cp = copy.copy(self)
+
+        if hasattr(self_cp, "_astra_db_collection"):
+            setattr(self_cp, "_astra_db_collection", None)
+
+        return _asdict(self_cp, **kwargs)
+
+    @property
+    @requires_dependencies(["astrapy"], extras="astra")
+    def astra_db_collection(self) -> "AstraDBCollection":
+        if self._astra_db_collection is None:
+            from astrapy.db import AstraDB
+
+            # Build the Astra DB object
+            self._astra_db = AstraDB(
+                api_endpoint=self.connector_config.access_config.api_endpoint,
+                token=self.connector_config.access_config.token,
+            )
+
+            # Create and connect to the newly created collection
+            self._astra_db_collection = self._astra_db.create_collection(
+                collection_name=self.connector_config.collection_name,
+                dimension=self.connector_config.embedding_dimension,
+                options={"indexing": {"deny": NON_INDEXED_FIELDS}},
+            )
+        return self._astra_db_collection
+
+    @requires_dependencies(["astrapy"], extras="astra")
+    @DestinationConnectionError.wrap
+    def initialize(self):
+        _ = self.astra_db_collection
+
+    @requires_dependencies(["astrapy"], extras="astra")
+    def check_connection(self):
+        try:
+            _ = self.astra_db_collection
+        except Exception as e:
+            logger.error(f"Failed to validate connection {e}", exc_info=True)
+            raise SourceConnectionNetworkError(f"failed to validate connection: {e}")
+
+    def write_dict(self, *args, elements_dict: t.List[t.Dict[str, t.Any]], **kwargs) -> None:
+        logger.info(f"Inserting / updating {len(elements_dict)} documents to AstraDB.")
+
+        astra_batch_size = self.write_config.batch_size
+
+        for chunk in chunk_generator(elements_dict, astra_batch_size):
+            self._astra_db_collection.insert_many(chunk)
+
+    def normalize_dict(self, element_dict: dict) -> dict:
+        return {
+            "id": str(uuid.uuid4()),
```

Comment:
Assuming i'm correct that this is the literal document that gets sent to Astra DB, i think let's change `id` to `_id`

Reply:
Oh. Is it not saving it in the native `id` of AstraDB? What is the actual schema of AstraDB?

Reply:
Got ya. Ideally we are putting the vector in the right place. We could completely remove `_id` if Astra DB makes it's own id.

Reply:
so where is it storing the text and metadata? We could name those whatever we want?

Reply:
Correct yep! The only one with the special name is `$vector` if the embeddings have been pre-computed (we're exploring computing them at insert time as an option, but right now that's not in place and we tend to rely on integrations like this to handle that.)

I'm going to be honest i'm a bit less sure about `_id` - but just testing it, it does seem like Astra generates that field automatically with a random uuid if not provided. so i think its safe to remove it if desired.